qa/workunits/rbd/test_admin_socket.sh

```diff
@@ -0,0 +1,142 @@
+#!/bin/bash -ex
+
+TMPDIR=/tmp/rbd_test_admin_socket$$
+mkdir $TMPDIR
+trap "rm -fr $TMPDIR" 0
+
+. $(dirname $0)/../ceph-helpers.sh
+
+function expect_false()
+{
+    set -x
+    if "$@"; then return 1; else return 0; fi
+}
+
+function rbd_watch_out_file()
+{
+    echo ${TMPDIR}/rbd_watch_$1.out
+}
+
+function rbd_watch_pid_file()
+{
+    echo ${TMPDIR}/rbd_watch_$1.pid
+}
+
+function rbd_watch_fifo()
+{
+    echo ${TMPDIR}/rbd_watch_$1.fifo
+}
+
+function rbd_watch_asok()
+{
+    echo ${TMPDIR}/rbd_watch_$1.asok
+}
+
+function rbd_get_perfcounter()
+{
+    local image=$1
+    local counter=$2
+
+    ceph --admin-daemon $(rbd_watch_asok ${image}) perf dump |
+	jq ".\"librbd--rbd/${image}\".${counter}"
```

Comment:
Thanks, I will fix jq query.

The '--format xml and xmlstarlet' is the first thing I tried. Unfortunately it does not work due to "/" in "librbd--rbd/testimg23790" tag:

 zhuzha:~/ceph/ceph/src% ceph --format xml --admin-daemon out/client.admin.13067.asok perf dump |  xmlstarlet sel -t -m "."
 -:1.37: error parsing attribute name
 <perfcounter_collection><librbd--rbd/testimg23790><rd>0</rd><rd_bytes>0</rd_byte
                                     ^

(I think I have to create a PR for this).